### PR TITLE
Upgrade docker and travis to use Python 3.7

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description
The goal of this PR is to upgrade the development docker image and travis ci to python 3.7.
*Note: This is temporarily using the 3.7 RC since docker hub hasn't been updated since the official release, but reflects the same 3.7.x.*

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
python, be

##### Testing
local

### CODE REVIEW

**Comment by mbeacom** [comment]
Blocked by: https://github.com/travis-ci/travis-ci/issues/9069

File: app/app/travis.env [hunk 1]
```diff
@@ -1,3 +1,3 @@
 CACHE_URL=dbcache://my_cache_table
-DATABASE_URL=psql://postgres:postgres@localhost:5432/postgres
+DATABASE_URL=psql://postgres:postgres@localhost/testdb
```
Comment (owocki) [inline app/app/travis.env]: we'll need to update `restore_backup.bash` and `psql.bash` etc

**Comment by owocki** [comment]
did not test, but looks sane to me.

how do we want to roll out?

when do we want to upgrade to 3.7 on the prod / stage servers?

File: app/app/travis.env (hunk 1 above)
Comment (mbeacom) [reply app/app/travis.env]: @owocki This only affects the `travis` environment.

**Comment by mbeacom** [comment]
@owocki I can backup the volumes and upgrade Python this evening.

**Comment by owocki** [comment]
sounds good to me; are there any breaking changes to be aware of / do we need to regression test?

**Comment by mbeacom** [comment]
@owocki There shouldn't be, no.  Everything seems to be functioning appropriately on stage.  Should be good to give a review and possibly merge. cc @SaptakS @thelostone-mc

File: .travis.yml [hunk 2]
```diff
@@ -62,6 +65,3 @@ jobs:
         - docker push gitcoin/web:${TRAVIS_BUILD_NUMBER}
         - docker tag gitcoin/web:${TRAVIS_BUILD_NUMBER} gitcoin/web:stable
         - docker push gitcoin/web:stable
-
-addons:
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply .travis.yml]: Primarily to pin the psql version to `9.6`

**Pull request merged**